Graph Panel: New options for X-axis Min & Max (for histograms)

papagian opened the pull request:
Adds support for #8148

papagian committed:
Optionally set histogram x-axis min/max

torkelo commented on public/app/plugins/panel/graph/graph.ts:
```diff
@@ -334,8 +334,8 @@ class GraphElement {
         if (this.data.length) {
-          const histMin = _.min(_.map(this.data, s => s.stats.min));
-          const histMax = _.max(_.map(this.data, s => s.stats.max));
+          const histMin = panel.xaxis.min ? panel.xaxis.min : _.min(_.map(this.data, s => s.stats.min));
```
if panel.xaxis.min is 0 then this check is not working :)    (A common mistake we have done many times in Grafana)

papagian replied:
As a matter of fact, that is true. I prepare a fix.

papagian committed:
Fix histogram xaxis min/max tests

papagian committed:
Change xaxis min and max form input types to number

papagian committed:
Fix histogram x-axis min/max

This commit:
*fixes histogram computation when zero x-axis min/max are set
(they used to be ignored).
*validates the user defined x-axis min/max values
and ignores them if they result in invalid bucket sizes.

papagian committed:
Minor fix in values to histogram conversion

Filter out values outside the min and max boundaries
because they are assigned to uninitialized buckets
(outside min and max bounds).

papagian commented:
This seemingly innocent modification has been proved to be challenging.
The reason is that extra effort was required to be put on validating the user provided values for x-axis min and max because they are used in the bucket size computation and can result in zero or negative values.
The fix resolves these issues by ignoring the user defined x-axis min and max values in those cases but this has the obvious drawback that the user is never informed about this.
A better approach, maybe, would be to prevent the user from entering invalid values for x-axis min and max.
